### ISSUE TITLE
Process not ending when GUI closed

### ISSUE THREAD

**Issue by WHTaylor** [mantidproject/mantidimaging#365]
To reproduce:
 - Start the application
 - Close the application (either x out, or alt-f4)

After a keyboard interrupt, a stack trace is printed:
```
Exception ignored in: <module 'threading' from '/home/wtaylor/miniconda3/envs/cenv/lib/python3.7/threading.py'>
Traceback (most recent call last):
  File "/home/wtaylor/miniconda3/envs/cenv/lib/python3.7/threading.py", line 1308, in _shutdown
    lock.acquire()
  File "/home/wtaylor/miniconda3/envs/cenv/lib/python3.7/site-packages/socketio/client.py", line 25, in signal_handler
    return original_signal_handler(sig, frame)
  File "/home/wtaylor/miniconda3/envs/cenv/lib/python3.7/site-packages/engineio/client.py", line 41, in signal_handler
    return original_signal_handler(sig, frame)
```

**Comment by DTasev** [comment]
I think I have a fix for this in #325, the issue is socket_io thread doesn't die with the GUI thread. I added the disconnect in main_window `closeEvent`. I am not sure when I can wrap up that PR though, do you think I should pull out the fix in another one?

**Comment by WHTaylor** [comment]
It's a minor nuisance, so no particular rush as long as it gets fixed eventually.
